pkgs/cli_config/lib/src/provider.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+abstract class Provider {
```

Comment:
This class in particular I think has a very generic name, I would make it a `ConfigProvider` or something, and probably propagate that naming all the way up the class hierarchy.

Reply:
Is this in the public API?

As an internal detail, adding the prefix `Config` doesn't add new information over the package context.